src/python/pants/backend/docker/subsystem.py

```diff
@@ -0,0 +1,120 @@
+        defaults = []
+        for alias, registry in self.registries.items():
+            if registry.default:
+                defaults.append(alias)
+        if len(defaults) > 1:
+            raise ValueError(
+                "Multiple default Docker registries in the [docker].registries configuration: "
+                + ", ".join(defaults)
+                + "."
+            )
+
+    @classmethod
+    def from_dict(cls, d: dict[str, Any]) -> DockerRegistries:
+        return cls(
+            FrozenDict(
+                {alias: DockerRegistryOptions.from_dict(options) for alias, options in d.items()}
+            )
+        )
+
+    def __getitem__(self, alias_or_address: str | None) -> DockerRegistryOptions:
+        return cast(DockerRegistryOptions, self.get(alias_or_address, implicit_options=False))
+
+    def get(
+        self, alias_or_address: str | None, implicit_options: bool = True
+    ) -> DockerRegistryOptions | None:
+        if alias_or_address == "":
+            return None
+
+        if alias_or_address is not None and alias_or_address in self.registries:
+            return self.registries.get(alias_or_address)
+
+        # The registries are expected to be very few, so this for loop is not that expensive.
+        default: DockerRegistryOptions | None = None
+        for registry in self.registries.values():
+            if registry.address == alias_or_address:
+                return registry
+            if registry.default:
+                default = registry
+
+        if alias_or_address:
+            if implicit_options:
+                return DockerRegistryOptions(address=alias_or_address)
+
+            raise ValueError(f"Unknown Docker registry: {alias_or_address}.")
+
+        if default or implicit_options:
+            return default
+
+        raise ValueError("There is no default Docker registry configured.")
+
+
+class DockerOptions(Subsystem):
+    options_scope = "docker"
+    help = "Options for interacting with Docker."
+
+    @classmethod
+    def register_options(cls, register):
+        registries_help = (
+            dedent(
+                """\
+            Configure Docker registries. The schema for a registry entry is as follows:
+
+                {
+                    "registry-alias": {
+                        "address": "registry-domain:port",
+                        "default": bool,
+                    },
+                    ...
+                }
+
+            """
+            )
+            + (
+                "Only one registry may be declared as the default registry. If a registry value "
+                "is not provided in a `docker_image` target, the address of the default registry "
+                "will be used, if any.\n"
+                "The `docker_image.registry` may be provided with either the registry address or "
+                "the registry alias, or `None` if the image should not be associated with a custom "
+                "registry."
```

Comment:
The pr description indicates that you can use the empty string to disable the default registry, but I don't see code to do that, and it's not mentioned here. Is the description stale?

Reply:
Good catch. It is this help text that has gotten stale. None is the default, so it is empty string it should mention instead for disabling the use of the default registry.

The code for it is in docker/subsystem:DockerRegistries.get.